### ISSUE TITLE
Brexit and impact on intrastat

### ISSUE THREAD

**Issue by alexis-via** [OCA/intrastat-extrastat#117]
I've read in detail the official documentation for the intrastat declaration for France (DEB) updated on January 5 2021 with the details for Brexit:
https://www.douane.gouv.fr/la-douane/informations/bulletins-officiels-des-douanes/da/21-003  (in French only)
and I have also exchanged with the French fiscal administration to have some details about the services (DES).
Here are the requirements, according to the French official documentation:
1) If we ship some goods from France to Great Britain (UK without Northern Ireland), we should not mention it on intrastat
2) If we ship some goods from France to Northern Ireland, the EU VAT regime still applies and the sale should be declared in the intrastat declaration. Companies in Northern Ireland now have a new VAT number starting with XI (VIES has been updated).
3) if we sell a service in B2B without VAT to a company in the UK, we should NOT mention it in the intrastat declaration, even when the customer is in Northern Ireland. It means the VAT regime for Northern Ireland is different between goods and services !
4) When the country of origin/destination is Northern Ireland, we are supposed to use the code XI as country code.
5) For the Country of origin of the goods (Made in xxx), we should distinguish between Northern Ireland (using country code XI) and Great Britain (using country code XU)

Do you agree on the requirements ? If yes, we can continue the discussion and talk about the implementation.

**Comment by alexis-via** [comment]
Odoo v14 doesn't accept yet the new VAT numbers for Northern Ireland cf my bug report https://github.com/odoo/odoo/issues/64891

**Comment by jbaudoux** [comment]
@alexis-via 
Here are the changes I did for EU companies:
* Add in countries a new country: "Nothern Ireland" (code XI). Mark it to be part of the intrastat declaration. Add in "Europe" definition for the Intracom.
* Reconfigure "United Kingdom": Unmark to be part of intrastat declaration. Update all partners having this country to change their fiscal position to Extracom.

**Comment by alexis-via** [comment]
@jbaudoux Yes, and it's a very bad idea : you will have problems when generating a SEPA wire transfer or a transport label for these partners in northern Ireland, because your bank and/or the transporter will not accept this country code XI. For me, this is not the right solution.

**Comment by alexis-via** [comment]
Had a talk with @luc-demeyer today and he agrees on the requirements. We decided an implementation that is now implemented in intrastat_base, intrastat_product, l10n_fr_intrastat_product and l10n_fr_intrastat_service.

No revolution, the most important thing is still the fiscal position of the invoice and the "intrastat" property on this fiscal position. Companies based in Northern Ireland should have an intrastat fiscal position (needed for goods) and, in the service declaration, we have a check country_id.code != 'GB' to exclude those invoices. For source/dest country, the replacement of country_id.code == 'GB' by 'XI' is made in the XML export. Several warnings implemented in the generation of lines catch your attention if you have an intrastat fiscal position with country_id.code == 'GB' and the partner's VAT number doesn't starts with "XI".
